R/deployments.R

```diff
@@ -145,6 +145,9 @@ deploymentRecord <- function(name,
     url = url %||% "",
     version = version
   )
+  # convert any multi-value metadata entries into comma-separated values
+  # this prevents write.dcf from writing multiple records into one file.
+  metadata <- lapply(metadata, function(v) paste0(v, collapse = ", "))
```

Comment:
I don't see an equivalent change in the parsing code, so I assume that already existed?

Reply:
Also can now simplify line 141?

Reply:
I'm rewriting only `metadata` values, as those are not a well-specified part of the `rsconnect::deployApp()` contract. Additionally, I'm leaving the rewritten result as-is because I didn't want to try to grok the expectations of the caller.

I could be convinced otherwise (on both of your points).

Reply:
Ooooh, got it.

It does feel asymmetric to me to handle writing but not reading. I think it would be more symmetric to either error if metadata contains anything other that single strings on write, or to also parse comma separated values on read.

Reply:
I agree (symmetry is preferred), but the IDE and the Quarto R package already provide vectors of engines...

Reply:
In that case, the current code seems like the least worst solution to me 😄

Reply:
> In that case, the current code seems like the least worst solution to me 😄

@hadley - to confirm: the change in this PR is the least-worst solution, not the current `main` implementation, correct?